### PR TITLE
Tabs Component

### PR DESCRIPTION
**Pull request by jfrej**
**Backwards Compatibility Implications**

_None_

**New Features**

- `Tabs` component
- `IconMore` component

**Bug Fixes**

_None_

**Miscellaneous**

- Updated `Button` component colours

### CODE REVIEW

File: src/components/Tabs/Tabs.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import StyleObjectPropType from '../../prop-types/style';
+import TabsNav from './TabsNav';
+import TabsPanes from './TabsPanes';
+import TabsPane from './TabsPane';
+import TabsTab from './TabsTab';
+
+let lastInstanceId = 0;
+
+const propTypes = {
+    selectedIndex: PropTypes.number,
```
Comment (mattpauldavies) [inline src/components/Tabs/Tabs.js]: Could we sort this prop list alphabetically like the other components please :)

File: src/components/_variables.scss [hunk 2]
```diff
@@ -8,6 +8,7 @@ $font-family: "DINOT", sans-serif !default;
 $white: #fff;
 $grey: #ccc !default;
 $charcoal: #3a333c !default;
```
Comment (mattpauldavies) [inline src/components/_variables.scss]: Let's check with Antoine although I think `charcoal` could be replaced with `slate` in all components

**Comment by mattpauldavies** [comment]
Overall, the code looks really good, just a couple of little comments above...

I checked in Firefox and there are some odd issues with the drop-down:

screen shot 2018-02-01 at 14 38 37

File: src/components/Tabs/Tabs.js (hunk 1 above)
Comment (jfrej) [reply src/components/Tabs/Tabs.js]: Oh, fixed!

**Comment by mattpauldavies** [comment]
Apart from agreeing which version of grey we want to use this all looks excellent.

Just one question: did you want to implement that trick you found to stop the menu nudging as it swaps between bold and normal?

Could we add that as an improvement to come back to later?

File: src/components/Tabs/TabsPane.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    relatedTabId: null,
+};
+
+class TabsPane extends Component {
+    state = {
+        isDeselecting: false,
+        isSelecting: false,
+    };
+
+    componentWillReceiveProps(nextProps) {
```
Comment (mattpauldavies) [inline src/components/Tabs/TabsPane.js]: `componentWillReceiveProps` is a little verbose... could this work as a more concise option:
```
componentWillReceiveProps(nextProps) {
    this.setState({
        isDeselecting: this.props.isSelected && !nextProps.isSelected,
        isSelecting: !this.props.isSelected && nextProps.isSelected,
    });
}
```

Comment (jfrej) [reply src/components/Tabs/TabsPane.js]: oh, yes, good point!

Comment (jfrej) [reply src/components/Tabs/TabsPane.js]: Fixed

File: src/components/Icon/IconMore.test.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+/* global describe, expect, it, shallow */
+import React from 'react';
+import IconArrow from './IconMore';
+
+describe('IconMore', () => {
+    it('has the appropriate classes', () => {
```
Comment (andyberry88) [inline src/components/Icon/IconMore.test.js]: Can we have a test to verify `render()` returns SVG content? (see https://github.com/Mudano/ui-react/pull/47#discussion_r166914332). We probably need to add these tests for each icon component.

Comment (jfrej) [reply src/components/Icon/IconMore.test.js]: Test added.
I've also created a helper for that which I'll include in a separate PR.

Comment (andyberry88) [reply src/components/Icon/IconMore.test.js]: 👍

File: src/components/Tabs/Tabs.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+        lastInstanceId += 1;
+
+        this.state = {
+            selectedIndex: this.props.selectedIndex,
+        };
+    }
+
+    render() {
+        const children = React.Children.map(this.props.children, (child) => {
```
Comment (mattpauldavies) [inline src/components/Tabs/Tabs.js]: I wonder if we should split this into multiple functions to make it easier to read and maintain?
Perhaps:
`getChildren()`
Which calls:
`getTabsNav()`
`getTabsPane()`

Comment (jfrej) [reply src/components/Tabs/Tabs.js]: Changed

**Pull request merged**